bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/_rtensor.py

```diff
@@ -9,10 +9,84 @@
 ################################################################################
 
 from .. import pythonization
-from .._rvec import _array_interface_dtype_map
+from .._rvec import _array_interface_dtype_map, _get_cpp_type_from_numpy_type
 from libROOTPythonizations import GetEndianess, GetDataPointer, GetSizeOfType
 import cppyy
 
+def _AsRTensor(arr):
+    """
+    Adopt memory of a Python object with array interface using an RTensor.
+
+    \param[in] self Always null, since this is a module function.
+    \param[in] obj PyObject with array interface
+
+    This function returns an RTensor which adopts the memory of the given
+    PyObject. The RTensor takes the data pointer and the shape from the array
+    interface dictionary.
+    """
+    import ROOT
+    import cppyy
+    import math
+    import platform
+
+    # Get array interface of object
+    interface = arr.__array_interface__
+
+    # Get the data-pointer
+    data = interface["data"][0]
+
+    # Get the size of the contiguous memory
+    shape = interface["shape"]
+    size = math.prod(shape) if len(shape) > 0 else 0
+
+    # Get the typestring and properties thereof
+    typestr = interface["typestr"]
+    if len(typestr) != 3:
+        raise RuntimeError(
+            "Object not convertible: __array_interface__['typestr'] returned '"
+            + typestr
+            + "' with invalid length unequal 3."
+        )
+
+    if typestr[:1] != GetEndianess():
+        raise RuntimeError(
+            "Object not convertible: Endianess of __array_interface__['typestr'] does not match endianess of ROOT."
+        )
+
+    dtype = typestr[1:]
+    dtypesize = int(typestr[-1])
+    cppdtype = _get_cpp_type_from_numpy_type(dtype)
+
+    # Get strides
+    strides = arr.strides
+
+    # Infer memory layout from strides
+    row_major = True
+    if len(strides) > 1 and strides[0] < strides[-1]:
+        row_major = False
+
+    # Construct an RTensor of the correct data-type
+    prefix = "0x" if platform.system() == "Windows" else ""
+    klassname = f"TMVA::Experimental::RTensor<{cppdtype},std::vector<{cppdtype}>>"
+    code = f"new {klassname}(reinterpret_cast<{cppdtype} *>({prefix}{data}),"
+    code += "{" + ",".join([f"{s}" for s in shape]) + "}, "
+    code += "{" + ",".join([f"{s // dtypesize}" for s in strides]) + "}, "
+    if row_major:
+       code += "TMVA::Experimental::MemoryLayout::RowMajor"
+    else:
+       code += "TMVA::Experimental::MemoryLayout::ColumnMajor"
+    code += ")"
```

Comment:
Bit of a nitpick, but I believe a multiline f-string would be more idiomatic here

Reply:
Sure, why not!